main.go

```diff
@@ -137,14 +158,19 @@ func main() {
 		case <-ticker:
 			fmt.Printf("%v runs so far, %v failures, over %s\n",
 				runs, fails, roundToSeconds(time.Since(startTime)))
+		case <-ctx.Done():
+			return ctx.Err()
 		}
 	}
-	fmt.Printf("%v runs completed, %v failures, over %s\n",
-		runs, fails, roundToSeconds(time.Since(startTime)))
-	if fails > 0 {
+}
+
+func main() {
+	switch err := run(); err {
+	case nil, context.DeadlineExceeded:
```

Comment:
Never mind, I understand. A comment would be helpful to indicate that this catches the case where stress saw no failures in the allotted time, not where a subprocess timed out.